x-pack/plugins/runtime_fields/public/components/runtime_field_form/runtime_field_form.tsx

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { useEffect } from 'react';
+import { i18n } from '@kbn/i18n';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSpacer,
+  EuiFormRow,
+  EuiComboBox,
+  EuiComboBoxOptionOption,
+  EuiLink,
+} from '@elastic/eui';
+
+import { useForm, Form, FormHook, UseField, TextField, CodeEditor } from '../../shared_imports';
+import { RuntimeField } from '../../types';
+import { RUNTIME_FIELD_OPTIONS } from '../../constants';
+
+import { schema } from './schema';
+
+export interface FormState {
+  isValid: boolean | undefined;
+  isSubmitted: boolean;
+  submit: FormHook<RuntimeField>['submit'];
+}
+
+export interface Props {
+  docsBaseUri: string;
+  defaultValue?: RuntimeField;
+  onChange?: (state: FormState) => void;
+}
+
+const RuntimeFieldFormComp = ({ defaultValue, onChange, docsBaseUri }: Props) => {
+  const { form } = useForm<RuntimeField>({ defaultValue, schema });
+  const { submit, isValid: isFormValid, isSubmitted } = form;
+
+  useEffect(() => {
+    if (onChange) {
+      onChange({ isValid: isFormValid, isSubmitted, submit });
+    }
+  }, [onChange, isFormValid, isSubmitted, submit]);
+
+  return (
+    <Form form={form}>
+      <EuiFlexGroup>
+        {/* Name */}
+        <EuiFlexItem>
+          <UseField path="name" component={TextField} data-test-subj="nameField" />
+        </EuiFlexItem>
+
+        {/* Return type */}
+        <EuiFlexItem>
+          <UseField<EuiComboBoxOptionOption[]> path="type">
+            {({ label, value, setValue }) => {
+              if (value === undefined) {
+                return null;
+              }
+              return (
+                <>
+                  <EuiFormRow label={label} fullWidth>
+                    <EuiComboBox
```

Comment:
I'm not as familiar with the proposed design - but would the `EuiSelect` suffice here, as there are only 6 supported types?

Reply:
For now, I've mirrored what CJ did for the mappings editor "runtime" field type in https://github.com/elastic/kibana/pull/79940
I'm happy to revisit this later if we don't think the ComboBox brings value 👍